Sources/Fluent/Query/Join.swift

```diff
@@ -18,10 +18,34 @@ public struct Join {
     /// the local data
     public let foreign: Entity.Type
```

Comment:
The difference in SQL:

### A
```sql
SELECT * FROM users
    JOIN pets
        ON pets.user_id = users.id
```

### B
```sql
SELECT * FROM users
    JOIN pets
        ON pets.id = users.pet_id
```

"users" is still the "local" table in the sense that it is the table we are joining into. Pets is the table being added into the equation. Left and right isn't a great name b/c sql has different joins like LEFT JOIN RIGHT JOIN and i don't know if that would conflict or be confusing.

![img_7630](https://cloud.githubusercontent.com/assets/1342803/23068096/813866c6-f522-11e6-8d50-4f1b52663a9a.JPG)

Here's a picture of the two different cases the join can have.